hack/generator/pkg/codegen/pipeline_remove_type_aliases.go

```diff
@@ -49,7 +49,8 @@ func removeTypeAliases() PipelineStage {
 func resolveTypeName(visitor *astmodel.TypeVisitor, name astmodel.TypeName, types astmodel.Types) (astmodel.Type, error) {
 	def, ok := types[name]
 	if !ok {
-		panic(fmt.Sprintf("Couldn't find type for type name %s", name))
+		klog.Warningf("Couldn't find type for type name %s", name)
```

Comment:
Why isn't this a panic or error? Isn't a name that points to a type not in the collection "bad" (bound to cause failures eventually?)

Reply:
Thinking more on this, are there cases where this isn't an error because it's a reference to a non-local type? If so possibly we should specifically check for that here by examining the `name.PackageReference`?

Reply:
Making it an error makes sense.

Reply:
I don't think it hurts too much for this to be a `panic` (though I have no problem with it as an error either).

Applying the "can the caller handle this" litmus test, I am not sure (in the context of our application) they can, which would leave a `panic` ok? (@porges has more thoughts on this than I do though)